File: api/src/endpoint/user/get_user.rs
use serde::{Deserialize, Serialize};
use api_derive::EndpointRequest;

use crate::{endpoint::Endpoint, resource::user::UserResource};

pub struct GetUser;

impl Endpoint for GetUser {
    const PATH: &'static str = "/user/{user_name}";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetUserRequest;
    type Response = GetUserResponse;
}

#[derive(EndpointRequest, Debug, Serialize, Deserialize)]
pub struct GetUserRequest {}

pub type GetUserResponse = UserResource;

File: gitdot-api-ts/src/endpoint/user/get-user.ts
import { z } from "zod";

import { UserResource } from "../../resource";

export const GetUserRequest = z.object({});
export type GetUserRequest = z.infer<typeof GetUserRequest>;

export const GetUser = {
  path: "/user/{user_name}",
  method: "GET",
  request: GetUserRequest,
  response: UserResource,
} as const;
export type GetUser = typeof GetUser;
